app/services/goldencobra/settings_cleanup.rb

```diff
@@ -0,0 +1,92 @@
+module Goldencobra
+  # This class offers methods for helping to clean up Golden Cobra settings. They
+  # can be used all over the project and in some included gems. Some of them are
+  # not used anymore but still appear in the backend to edit. This produces overhead
+  # because they suggest to be important and one will fill them.
+  # If we remove the unused settings, nobody needs to care about them.
+  class SettingsCleanup
+    # Checks every single Goldencobra::Setting for presence or if it is still used.
+    #
+    # @param settings_path [string] the path of the settings.yml
+    #        Examples: Goldencobra::Engine.root + "config/settings.yml"
+    #                  GoldencobraEvents::Engine.root + "config/settings.yml"
+    def self.remove_unused_setting_keys(settings_path)
+      raise "Settings file '#{path_file_name}' does not exist" unless File.exist?(settings_path)
+
+      settings = open(settings_path) { |f| YAML.safe_load(f) }
+      settings_root = settings.each_key.first if settings.present?
+
+      return if settings_root.blank?
+
+      p "*" * 100
+      Goldencobra::Setting.roots.where(title: settings_root).first.descendants.each do |setting|
+        settings_to_fetch = settings
+        setting.path.pluck(:title).each do |key|
+          p "Setting key: #{key} - Setting path: #{setting.path.pluck(:title).join(".")}"
+          begin
+            # Traverse the setting path hierarchically from the outside into the deep.
+            #
+            # Example:
+            #   setting path = goldencobra.locations.geocoding
+            #
+            #   1. settings_to_fetch = {"goldencobra"=>{"locations"=>{"geocoding"=>"true"}}}
+            #   2. settings_to_fetch = {"locations"=>{"geocoding"=>"true"}}
+            #   3. settings_to_fetch = {"geocoding"=>"true"}
+            settings_to_fetch = settings_to_fetch.fetch(key)
+          rescue KeyError
+            begin
+              # If a 'KeyError' is thrown, the key is not present and the setting can be
+              # removed from the database.
+              p "Not present!"
+              p "Destroy setting: #{setting.title}"
+              p "-" * 100
+              setting.destroy
+            rescue Ancestry::AncestryException
+              # If a `Ancestry::AncestryException` is thrown, than do not quit the
+              # process because there could be some more settings to check in the loop.
+              # Go on with the next iteration step.
```

Comment:
I think the exception raises when the setting is not processable. So it is also not destroyable.